### PR TITLE
Migrate to buf v2 config and update deps

### PR DESCRIPTION
**Pull request by majst01**
Breaking for client, client.Dial does not require a context anymore

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func init() {
 // longer than the actual operation context timeout (as opposed to the `ctx`
```
Comment (robertvolkmann) [inline client.go]: Please remove references to `ctx` from the comment block.

**Comment by robertvolkmann** [comment]
According https://github.com/grpc/grpc-go/issues/1786, there a couple of difference between `Dial` and `NewClient`. Do observed any impact of these changes?

**Comment by majst01** [comment]
> According [grpc/grpc-go#1786](https://github.com/grpc/grpc-go/issues/1786), there a couple of difference between `Dial` and `NewClient`. Do observed any impact of these changes?

DialContext is deprecated

**Comment by majst01** [comment]
> According [grpc/grpc-go#1786](https://github.com/grpc/grpc-go/issues/1786), there a couple of difference between `Dial` and `NewClient`. Do observed any impact of these changes?

In the duros-controller i call the Version Enpoint and the duros-controller will die and restart if the client is not working.

https://github.com/metal-stack/duros-controller/blob/master/main.go#L226

**Review by robertvolkmann — APPROVED** [review]

**Pull request merged**